### ISSUE TITLE
Alignment of the bm network

### ISSUE THREAD

**Issue by Inosonnia** [cvlab-stonybrook/DewarpNet#29]
I implemented the two-phase training schedule by exploiting the provided doc3D dataset (~10w).
However, the output provided by infer.py is not correct and it seems that the networks are not properly trained.

To pinpoint the problem, I download the provided official pre-trained models.
With official wc network + official bm network, the results are pretty good.
With my trained wc network + official bm network, the results look fine.
With my bm network + official wc network, the result is not correct, and it is the problem.

The loss of training bm seems fine (with lr = 1e-4, the train and val MSE loss decreases continuously and approximate 5 * 1e-5 at epoch 50).
The mentioned solutions in other issues like pytorch version (pt1.4 + pt0.4.1) are tried but do not help.

The only modification is in doc3dbmnoimgc_loader.py, line 29, the self.altroot = "xxx/swat3d".
Since I can not find this folder in the project, it is altered to self.root according to my understanding.

Would you please point out other possible reasons?

Looking forward for your reply, thanks!

**Comment by sagniklp** [comment]
What is the problem? 
Could you please share the results with your bm network + official wc network?

**Comment by Inosonnia** [comment]
I start the train procedure by this command, with pytorch 0.4.1:
python trainbm.py --arch dnetccnl --img_rows 128 --img_cols 128 --n_epoch 100 --batch_size 16 --l_rate 0.0001 --data_path ./doc3d-data

After training 75 epochs, I have: 

Epoch[75/100] Batch [6350/6379] Loss: 0.0641
Training L1:0.005895
Training MSE:'6.03227600935e-05'

The attached image is the inferred result of ./eval/inp/14_2.png.
Other example images share similar performance.

![14_2](https://user-images.githubusercontent.com/11433902/104689233-22005200-573d-11eb-8e4e-ed811eb512cc.jpg)

**Comment by sagniklp** [comment]
I think there's something wrong with the input to the BM. When you train the BM network do you normalize WC? Use the same normalization while testing.

**Comment by Inosonnia** [comment]
It is indeed the normalization issue.
Problem solved, thanks!
